cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/ExcludedBrokersForLeadershipTest.java

```diff
@@ -90,11 +86,23 @@
     // With single excluded broker, rack aware unsatisfiable cluster, one dead broker (Exception)
     p.add(params(5, RackAwareGoal.class, excludeB1, OptimizationFailureException.class, DeterministicCluster.rackAwareUnsatisfiable(), deadBroker0, null));
 
+    // ============ReplicaCapacityGoal============
+    // Test: With single excluded broker, satisfiable cluster, no dead brokers (No exception, No proposal
+    // for excluded broker, Expected to look optimized)
+    p.add(params(0, ReplicaCapacityGoal.class, excludeB1, null, unbalanced(), noDeadBroker, true));
+    // Test: With single excluded broker, satisfiable cluster, one dead brokers (No exception, No proposal
+    // for excluded broker, Expected to look optimized)
+    p.add(params(1, ReplicaCapacityGoal.class, excludeB1, null, unbalanced(), deadBroker0, true));
+    // Test: With all brokers excluded, no dead brokers, not satisfiable (No exception, No proposal
+    // for excluded broker, Expected to look optimized)
+    p.add(params(2, ReplicaCapacityGoal.class, excludeAllBrokers, null, unbalanced(), noDeadBroker, true));
```

Comment:
This is similar with the case of CpuCapacityGoal, since we originally set the per-broker replica count to 1, this goal can hardly be satisfied and always throw exception.